pjsip/src/pjsip/sip_auth_client.c

```diff
@@ -712,6 +737,26 @@ static pjsip_cached_auth *find_cached_auth( pjsip_auth_clt_sess *sess,
                                             const pj_str_t *realm,
                                             pjsip_auth_algorithm_type algorithm_type)
 {
+#if defined(PJSIP_SHARED_AUTH_SESSION) && PJSIP_SHARED_AUTH_SESSION
+    pj_status_t lock_status;
+    pj_bool_t with_parent = PJ_FALSE;
+    lock_status = pj_lock_acquire(sess->lock);
+    pjsip_cached_auth * pauth = NULL;
+    if (lock_status != PJ_SUCCESS) {
+        return NULL;
+    }
+    if (sess->parent) {
+         pauth = sess->parent->cached_auth.next;
```

Comment:
No iteration to lookup the matched `realm` & `algorithm_type` on parent?
(Or perhaps recursive call to self like in `auth_find_cred()`)